apps/web/vibes/soul/sections/address-list-section/index.tsx

```diff
@@ -141,132 +127,146 @@ export function AddressListSection<A extends Address>({
-                }
-                onSubmit={(formData) => {
-                  setActiveAddressIds((prev) => prev.filter((id) => id !== address.id));
+            <div className="w-[480px] space-y-4">
+              <DynamicForm
+                action={(_prevState, formData) => {
+                  setShowNewAddressForm(false);
 
                   startTransition(() => {
                     formAction(formData);
                     setOptimisticState(formData);
                   });
+
+                  return {
+                    fields: optimisticState.fields,
+                    lastResult: optimisticState.lastResult,
+                  };
                 }}
-                phoneLabel={phoneLabel}
-                postalCodeLabel={postalCodeLabel}
-                submitLabel={updateLabel}
-              />
-            ) : (
-              <div className="space-y-4">
-                <AddressPreview
-                  address={address}
-                  isDefault={
-                    optimisticState.defaultAddress
-                      ? optimisticState.defaultAddress.id === address.id
-                      : undefined
+                buttonSize="small"
+                cancelLabel={cancelLabel}
+                fields={optimisticState.fields.map((field) => {
+                  if ('name' in field && field.name === 'id') {
+                    return {
+                      ...field,
+                      id: 'new',
+                    };
                   }
-                />
-                <div className="flex gap-1">
-                  <Button
-                    aria-label={`${editLabel}: ${address.firstName} ${address.lastName}`}
-                    onClick={() => setActiveAddressIds((prev) => [...prev, address.id])}
-                    size="small"
-                    variant="tertiary"
-                  >
-                    {editLabel}
-                  </Button>
-                  {optimisticState.addresses.length > minimumAddressCount && (
-                    <AddressActionButton
-                      action={formAction}
-                      address={address}
-                      aria-label={`${deleteLabel}: ${address.firstName} ${address.lastName}`}
-                      intent="delete"
-                      onSubmit={(formData) => {
-                        startTransition(() => {
-                          formAction(formData);
-                          setOptimisticState(formData);
-                        });
-                      }}
-                    >
-                      {deleteLabel}
-                    </AddressActionButton>
-                  )}
 
-                  {optimisticState.defaultAddress &&
-                    optimisticState.defaultAddress.id !== address.id && (
+                  return field;
+                })}
+                onCancel={() => setShowNewAddressForm(false)}
```

Comment:
We should look into using conform's "intent" feature so that we don't need to use a callback that will force this to be a client component. We can still get instant behavior using optimistic state